src/main/java/org/sonarsource/sonarlint/ls/settings/RulesConfiguration.java

```diff
@@ -94,8 +94,19 @@ private static String safeParseLevel(Map<String, Object> config) {
       Map<String, Object> parameters = parametersValue instanceof Map ? (Map<String, Object>) parametersValue : Collections.emptyMap();
       return parameters.entrySet().stream()
         .filter(e -> e.getValue() != null)
-        .map(e -> new AbstractMap.SimpleImmutableEntry<>(e.getKey(), e.getValue().toString()))
+        .map(e -> new AbstractMap.SimpleImmutableEntry<>(e.getKey(), safeStringValue(e.getValue())))
         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
     }
+
+    private static String safeStringValue(Object paramValue) {
+      if (paramValue instanceof Double) {
```

Comment:
Are we confident about this? What if lsp4j decides to change to Float? Shouldn't we test `instanceof Number`?

Reply:
This is Gson's default behavior (from `org.sonarsource.sonarlint.ls.Utils#parseToMap`).

Reply:
Fine then